src/tools/rbd/action/MirrorPool.cc

```diff
@@ -1405,8 +1432,22 @@ int execute_info(const po::variables_map &vm,
     if (r < 0) {
       return r;
     }
+  } else {
+    r = rbd.mirror_remote_namespace_get(io_ctx, &remote_namespace);
+    if (r < 0 && r != -EOPNOTSUPP && r != -ENOENT ) {
+      return r;
+    }
+    if (r < 0) {
+      remote_namespace = namespace_name;
+    }
   }
 
+  if (mirror_mode != RBD_MIRROR_MODE_DISABLED) {
```

Comment:
I think it is best to get all the information up front so we don't error out in the middle of printing the output.

Reply:
Makes sense, although I'd note that errorring out in the middle of printing the output can still happen for pools because peer attributes aren't fetched in advance.